File: whisper/src/main/java/com/yy/mobile/whisper/Immutable.java
package com.yy.mobile.whisper;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author YvesCheung
 * 2018/9/8.
 *
 */
@Documented
@Target({ElementType.FIELD,
        ElementType.METHOD,
        ElementType.LOCAL_VARIABLE,
        ElementType.PARAMETER})
@Retention(RetentionPolicy.SOURCE)
public @interface Immutable {
}

File: whisper/src/main/java/com/yy/mobile/whisper/Immutable.kt
package com.yy.mobile.whisper


/**
 * @author YvesCheung
 * 2018/9/8.
 */
@MustBeDocumented
@Retention(AnnotationRetention.SOURCE)
@Target(
    AnnotationTarget.FIELD,
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
    AnnotationTarget.LOCAL_VARIABLE,
    AnnotationTarget.VALUE_PARAMETER)
annotation class Immutable
